src/components/link/link.js

```diff
@@ -115,26 +115,40 @@ export const computed = {
 }
 
 function computeTag (props, parent) {
-  return Boolean(parent.$router) && props.to && !props.disabled ? 'router-link' : 'a'
+  return (parent.$router && props.to && !props.disabled) ? (parent.$nuxt ? 'nuxt-link' : 'router-link') : 'a'
+}
+
+function isRouterLink (tag) {
+  return tag !== 'a'
 }
 
 function computeHref ({ disabled, href, to }, tag) {
   // We've already checked the parent.$router in computeTag,
-  // so router-link means live router.
-  // When deferring to Vue Router's router-link,
-  // don't use the href attr at all.
-  // Must return undefined for router-link to populate href.
-  if (tag === 'router-link') return void 0
+  // so isRouterLink(tag) means live router.
+  // When deferring to Vue Router's router-link, don't use the href attr at all.
+  // We must return undefined for router-link to properly populate href.
+  if (isRouterLink(tag)) {
+    return void 0
```

Comment:
I would suggest to use `undefined` here because it's used at most other places in the repo to and change the other few `void 0`'s to `undefined` too in a followup PR.

Reply:
We originally had `undefined` here, but had to fall back to `void 0` because of issues with `<router-link>`

Reply:
Although it is possible that updates to vue-router may have fixed this.

Reply:
Although maybe lines 216-219 fixed that by explicitly removing the `href` prop

```js
    // If href prop exists on router-link (even undefined or null) it fails working on SSR
    if (!componentData.attrs.href) {
      delete componentData.attrs.href
	  }
```

Reply:
After some digging and playing, the `void 0` is no longer needed.